src/Umbraco.Web.UI.Client/src/views/propertyeditors/blockeditor/blockcard/blockcard.component.js

```diff
@@ -0,0 +1,23 @@
+(function () {
+    "use strict";
+
+    angular
+        .module("umbraco")
+        .component("umbBlockCard", {
+            templateUrl: "views/propertyeditors/blockeditor/blockcard/blockcard.component.html",
```

Comment:
Many of these new components have view names that don't match the component name, for example this is umbBlockCard and normally would expect to be able to search for `umb-block-card.html` but that won't be found which makes it a little harder to figure out where the view is. I understand that now the view is next to the component! whoohoo! that is wonderful so maybe this doesn't matter but I just think consisitency is important so maybe it does?

Reply:
I'm definitely not a fan of the naming convention today or the structure. So I think the aim is to settle on a new structure and slowly get everything reorganized this way. This is sort of a test of how that structure could be. But if you feel strongly for keeping to the old naming style depending on their file format then we can change it?
From my perspective, this was a way to test out how it could work and then iterate from that.

Reply:
@nielslyngsoe 

> This is sort of a test of how that structure could be

Since we are releasing this publicly I don't really think we should release things in a 'test' state since we cannot undo it once it's out. The issue I'm seeing in this PR is that it's not following any naming convention. If there was a convention I'd be fine with it but there doesn't seem to be:

* umbBlockListBlockContent = blocklist.block.component.js
   * I'm unsure what the convention is trying to be but might suspect this should be named: blocklist.blockcontent.component.js ? I'm fine if we remove the umb prefix for components and keep the normal name of the component as the prefix but the rest of the file name should be consistent?
* blockListPropertyEditor = blocklist.component.js = blocklist.component.html
    * missing umb prefix which we still need to maintain consistency for for all directives/components (unfortunately this wasn't followed for nested content either)
    * should probably be renamed to umbBlockList? Do we need the PropertyEditor part? If so should the file name be blocklist.propertyeditor.component.js ?
* umbBlockListScopedBlockContent = blocklist.scopedblock.component.js
    * should the file name be blocklist.scopedblockcontent.component.js?
* "Umbraco.PropertyEditors.BlockListPropertyEditor.CreateButtonController" = blocklist.component.createButton.controller.js 
    * This one is tricky because it's a controller that belongs directly to the component, not sure the correct approach here but if it were me I would just add this controller into the component file
* umbElementEditorContent = elementEditor.content.component.js = elementEditor.content.component.html
   * Not really sure whats going on here, why is there a .content. ? should this be elementeditorcontent.component.js ?

2 of the views is odd: views/common/infiniteeditors/elementeditor/elementeditor.content.html & views/common/infiniteeditors/elementeditor/elementeditor.settings.html , these exist in this elementeditor folder but they are only ever rendered using the "Umbraco.Editors.BlockEditorController" controller which exists in a different infinite editors folder. I 'think' it makes more sense to have these 2 views as part of the same folder where they are used?

Reply:
> @nielslyngsoe
> 
> > This is sort of a test of how that structure could be
> 
> Since we are releasing this publicly I don't really think we should release things in a 'test' state since we cannot undo it once it's out. The issue I'm seeing in this PR is that it's not following any naming convention. If there was a convention I'd be fine with it but there doesn't seem to be:
> 
> * umbBlockListBlockContent = blocklist.block.component.js
>   
>   * I'm unsure what the convention is trying to be but might suspect this should be named: blocklist.blockcontent.component.js ? I'm fine if we remove the umb prefix for components and keep the normal name of the component as the prefix but the rest of the file name should be consistent?
> * blockListPropertyEditor = blocklist.component.js = blocklist.component.html
>   
>   * missing umb prefix which we still need to maintain consistency for for all directives/components (unfortunately this wasn't followed for nested content either)
>   * should probably be renamed to umbBlockList? Do we need the PropertyEditor part? If so should the file name be blocklist.propertyeditor.component.js ?
> * umbBlockListScopedBlockContent = blocklist.scopedblock.component.js
>   
>   * should the file name be blocklist.scopedblockcontent.component.js?
> * "Umbraco.PropertyEditors.BlockListPropertyEditor.CreateButtonController" = blocklist.component.createButton.controller.js
>   
>   * This one is tricky because it's a controller that belongs directly to the component, not sure the correct approach here but if it were me I would just add this controller into the component file

Agree. If this controller is only relevant to this component I'm okay appending it to the component file to keep things together.

> * umbElementEditorContent = elementEditor.content.component.js = elementEditor.content.component.html
>   
>   * Not really sure whats going on here, why is there a .content. ? should this be elementeditorcontent.component.js ?
> 
> 2 of the views is odd: views/common/infiniteeditors/elementeditor/elementeditor.content.html & views/common/infiniteeditors/elementeditor/elementeditor.settings.html , these exist in this elementeditor folder but they are only ever rendered using the "Umbraco.Editors.BlockEditorController" controller which exists in a different infinite editors folder. I 'think' it makes more sense to have these 2 views as part of the same folder where they are used?

Regarding naming conventions I do not care too much about how we name things. As long as it is consistent and follows something we can all agree on. Since I'm usually not much in the frontend/angular parts, I would rather that those who actually *are*, make this decision.

Personal preference though is that I like the approach of grouping files together component-wise instead of spreading out views and js and whatnot in completely different locations. So please make sure to stick with that 😄

And - as @Shazwazza points out, we should not be "testing" naming conventions to see what works in code that is going public. Especially not without specifically highlighting that we have done this and we need to make a decision before this is merged in. It's fine to _test_ conventions as it can be fairly hard to imagine the full scope of how something works without seeing it in action - but if this had not been caught we would have released this code with inconsistent naming making our codebase even worse in an area where we've been (and are) working on improving what we do.

Reply:
I will take the word testing back. What I meant was just, its the first time we group files together this way and I haven't had the time to come up with an overall plan for the whole project. But I think when putting HTML, CSS and JS in the same folder, the old naming convention of having the controller named with camelCase and the view named with dashes is a bit annoying.

But actually looking at the other property-editors this actually follows that naming convention quite well, so there is really not much new about this, except we also have the CSS file together with HTML and JS. — Its not just nested content but all property editors are named this way.

I have corrected a few minor mistakes as Shannon pointed out.

But otherwise, let's have a chat about it, I´m happy to change it, but please have a look at the other infinite editors and property editors since they are named similar.

Reply:
Okay, I see that the components aka directives that I have collected together in the property editor folder, could be named in our existing naming convention, views with dashed and components/directives as one long word with lowercase... My only concern regarding that is that this would then seem weird put next to each other in the same folder.

See this example for this infinite editor:
![image](https://user-images.githubusercontent.com/6791648/85791424-13ffcb00-b732-11ea-82a6-537dd8b95f5f.png)

Compared to how I have done it for now:
![image](https://user-images.githubusercontent.com/6791648/85791526-41e50f80-b732-11ea-9b52-d4e128a25917.png)

But I completely understand that the naming convention makes it possible for devs to search on the DOM-element-name and use that to find the view or directive behind it, so that would argue for the old naming convention though it looks weird put next to each other.

I can also take those components/directives out, but then we end up having files everywhere which I'm quite against — even though that's what we have today.:-)

Reply:
I have corrected for now, please have a look to see wether you feel its done right. Thanks

Reply:
@Shazwazza